02-peer-protocol.md

```diff
@@ -776,6 +788,83 @@ satoshis, which is possible if `dust_limit_satoshis` is below 546 satoshis).
 No funds are at risk when that happens, but the channel must be force-closed as
 the closing transaction will likely never reach miners.
 
+### Closing Negotiation: `closing_complete` and `closing_sig`
+
+Once shutdown is complete, the channel is empty of HTLCs, there are no commitments
+for which a revocation is owed, and all updates are included on both commitments,
+the final current commitment transactions will have no HTLCs.
+
+Each peer says what fee it will pay, and the other side simply signs that transaction.  The only complexity comes from allowing each side to omit its own output should it be uneconomic.
+
+This process will be repeated every time a `shutdown` message is received, which allows re-negotiation.
```

Comment:
Right now redundant `shutdown` messages are ignored: it won't be the case anymore (especially with taproot since we'll need to use the nonces from that message). Should we now allow the sender to use a different closing `scriptpubkey` than the one they used in previous `shutdown`? Or should we keep restricting it for simplicity?

Reply:
I guess allowing different `scriptpubkey` should be possible and the closer want to redirect the funds in another `scriptpubkey`.

ofc at this point we should also send it back inside the `closing_complete` msg, imho

Reply:
Yes, let them change it. Why not?

Reply:
I guess we should have this because we can have concurrency around the network? and getting the confirmation on what scriptpubkey we close the channel can be useful to avoid unnecessary mess,  right?

Reply:
No concurrency, reconnection requirements already say you re-xmit on every reconnect:

```
  - upon reconnection:
    - if it has sent a previous `shutdown`:
      - MUST retransmit `shutdown`.
```

Reply:
Ah I was missing it! good point, but in this case, the concurrency can happen if I send two subsequent shutdown messages with different `scriptpubkey`. I do not know why people should do it, but the spec allows it